### PR TITLE
SC structure roadmap

### PR DESCRIPTION
**Pull request by morrowcj**
## Summary
This PR is primarily meant to update the documents associated with the soil and crop module redesign. 

## Details
* The SC_redesign directory has been slightly reorganized. The details of the current structure can be found in the new `README_SC-redesign.md` file. The bulk of the changes are simply moving files around.
* The road map for the crop and soil redesign has been updated to reflect recent changes to the structure of the code, and to provide more detail. The main file affected is `road-map_soil-and-crop.md`. Mostly this involved
* Files have been updated with relative path links to allow docs to connect to each other easily.
* I also modified the shell script that checks for `TODO`s in the SC module (`find_SCredesign-TODOs.sh`).

## Guide for Reviewers
The "Files Changed" tab looks more complicated than it is because of moving files. The best path for reviewers, as I see it, is to 
1. look over `README_SC-redesign.md` first, 
2. then take a look at `road-map_soil-and-crop.md` - This file hasn't changed much, but it still would be good for other subject matter experts to review this document in its entirety to ensure it is clear. This file will be used guide our team and provide a path for someone to take over for me if needed.
3.  Look at the changes to `functionality-requirements.md`, if desired, but I just moved the sections around.
4. All other files were reviewed during fix-athon and have just been moved or have very minor changes.

### CODE REVIEW

**Review by JoeWaddell — APPROVED** [review]
This is looking great! I appreciate you tagging me on this, as @Ed731551 and I will be fleshing out the Animal Module sections in the coming weeks, and we'll certainly keep our style/layout consistent across modules.

**Comment by LaplacetteCM** [comment]
This looks good Clay. I think it is clearly explained and detailed to become a useful resource for S&C.

**Review by LaplacetteCM — APPROVED** [review]

**Comment by KFosterReed** [comment]
I started to review this tonight and will return in the morning. First impression is that I love it! I was skeptical about having the scientific documentation within the Git Repo but you have changed my mind! I think this is a new standard in documentation for the RuFaS team :)

**Comment by morrowcj** [comment]
> I was skeptical about having the scientific documentation within the Git Repo but you have changed my mind! I think this is a new standard in documentation for the RuFaS team :)

I'm really glad to hear that! Thank you! And if you ever change your mind, it is super easy to convert these text-only files to any other format.

File: SC_redesign/docs_SC/Admin/road-map_soil-and-crop.md [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+etc. The main classes `Field`, `Crop`, and `Soil` all utilize from such process classes. 
+The benefit of these process superclasses is that they can be run/tested in isolation while coming 
+together in the main class to work
+together for high-level processes. It also allows each of the units or classes to be completely agnostic of
+the other units. 
+
+### Main (composite) classes
+
+In general, the main composite classes should receive most of their methods and attributes
+from their respective components (e.g., `Crop` uses has an attribute `nitrogen_incorporation` which is an instance
```
Comment (Isgotkowitz) [inline SC_redesign/docs_SC/Admin/road-map_soil-and-crop.md]: Remove either "uses" or "has" from this line

**Review by Isgotkowitz — APPROVED** [review]
Is clear about what needs to be done and how it should be done, looks good to me

**Pull request merged**